Code/GraphMol/RGroupDecomposition/RGroupCore.cpp

```diff
@@ -816,4 +817,27 @@ int RCore::matchingIndexToCoreIndex(int matchingIndex) const {
                   "Matched atom missing core index");
   return atom->getProp<int>(RLABEL_CORE_INDEX);
 }
+
+// Create tautomer query for the matching mol on demand and cache for performance
+std::shared_ptr<TautomerQuery> RCore::getMatchingTautomerQuery() {
+  if (!checkedForTautomerQuery) {
+    try {
+      // Enumerate tautomers from a sanitized copy of the matching molecule
+      const auto copy = new RWMol(*matchingMol);
```

Comment:
I think this could just be 

```
RWMol copy(*matchingMol);
```

and not a pointer.

Reply:
I've made that change